tests/documentation/command-reference/doc_command_reference_init_test.go

```diff
@@ -0,0 +1,207 @@
+package docautomation
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-developer/odo/tests/helper"
+)
+
+var _ = Describe("doc command reference odo init", Label(helper.LabelNoCluster), func() {
+	var commonVar helper.CommonVar
+	var commonPath = filepath.Join("command-reference", "docs-mdx", "init")
+	var outputStringFormat = "```console\n$ odo %s\n%s```\n"
+
+	BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach()
+		helper.Chdir(commonVar.Context)
+		Expect(helper.VerifyFileExists(".odo/env/env.yaml")).To(BeFalse())
+	})
+
+	AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+	// interactive tests do not provide the same output every time,
+	// so we'll skip these tests until we have more coverage and then investigate a better way to test this
+	Context("Interactive Mode", func() {
+		It("Empty directory", func() {
+			args := []string{"odo", "init"}
+			out, err := helper.RunInteractive(args, []string{"ODO_LOG_LEVEL=0"}, func(ctx helper.InteractiveContext) {
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "Java")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, "my-java-maven-app")
+
+				helper.ExpectString(ctx, "Changes will be directly reflected on the cluster.")
+			})
+			Expect(err).To(BeNil())
+			got := helper.StripAnsi(out)
+			got = helper.StripInteractiveQuestion(got)
+			got = fmt.Sprintf(outputStringFormat, args[1], helper.StripSpinner(got))
+			want := helper.GetMDXContent(filepath.Join(commonPath, "interactive_mode_empty_directory_output.mdx"))
+			diff := cmp.Diff(want, got)
+			Expect(diff).To(BeEmpty())
```

Comment:
I think it would be very helpful to also display the diff if there is an error. It is possible to pass an optional description, like so;

```go
  Expect(diff).To(BeEmpty(), fmt.Sprintf("mismatch (-want +got):\n%s", diff))
```

Reply:
I do think what we have right now is pretty self sufficient but I'd like to understand this suggestion.
I am confused about the meaning of (-want +got), can you explain a bit?

Reply:
I was thinking about what we have in our unit tests, where we are displaying the diff when there is a mismatch (to help investigate the issue). And `(-want +got)` is just some convention used to display the diff.
But my bad, Gingko's `Expect` already displays the value if the matcher returns an error. So it's okay.